en/Case_study/yourFirstModule.md

```diff
@@ -35,8 +35,12 @@ To complete the `mvn openmrs-sdk:create-project` workflow, you will be prompted
 * **`What kind of project would you like to create?:`** You may choose to create a platform module, which can be run on any server or an OpenMRS Reference Application module, which needs to be run on a server with the OpenMRS Reference Application distribution installed. For this example, choose type `1`.
 * Next, you will be prompted for `module id`, `module name`, `description`, `groupId`, `initial version` and the lowest version of `platform support`. For this example, you can stick with the default provided values.
 
+
 For more info, see [module conventions](https://wiki.openmrs.org/display/docs/Module+Conventions).
 
+N.B. in the Module Activation class, The SDK will implement the deprecated ModuleActivator interface. To be able to support openmrs version 2+, u need to instead extend the BaseModuleActivator Class in order to activate your Module.
```

Comment:
the ModuleActivator interface is deprecated. and unfortunately thats wat the openmrs sdk implements. hence currently we just extend to BaseModuleActivator

Reply:
Can you point me to where you get that the ModuleActivator interface is deprecated?

Reply:
bya, but even before that , the IDE  informs u , because it crosses the "ModuleActivator" with a black line and informs u that the interface is deprecated

Reply:
```
A module Activator class MUST implement the interface 'org.openmrs.module.ModuleActivator'. There is an abstract class 'org.openmrs.module.BaseModuleActivator' that is optionally available as a default implementation and "buffer" against changes. We recommend extending the abstract class in your module for "forward compatibility" sake. These are available in version 1.7.0 and later.

If you need your module to support earlier versions up to 1.6.x, your activator class should implement the deprecated 'org.openmrs.module.Activator' which has only 2 methods. ModuleActivator gives more control to the developer with its 6 methods and so is preferred over the older and simpler Activator interface
```
thats from thatlink i quoted
https://wiki.openmrs.org/display/docs/Module+Activator

Reply:
```
A module Activator class MUST implement the interface 'org.openmrs.module.ModuleActivator'. There is an abstract class 'org.openmrs.module.BaseModuleActivator' that is optionally available as a default implementation and "buffer" against changes. We recommend extending the abstract class in your module for "forward compatibility" sake. These are available in version 1.7.0 and later.

If you need your module to support earlier versions up to 1.6.x, your activator class should implement the deprecated 'org.openmrs.module.Activator' which has only 2 methods. ModuleActivator gives more control to the developer with its 6 methods and so is preferred over the older and simpler Activator interface
```
thats from thatlink i quoted
https://wiki.openmrs.org/display/docs/Module+Activator

Reply:
The sentence you are referencing, which of the two does it state as deprecated? Activator or ModuleActivator?

Reply:
oh well , your actually right . and am wrong. just that i remeber at one time a module had failed to start  , just because it had implemented the ModuleActivator. and it only started wen i instead extended the BaseModuleActivator.

Reply:
lett me correct that